internal/image/docker.go

```diff
@@ -0,0 +1,88 @@
+package image
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const (
+	DefaultDockerHost      = "docker.io"
+	DefaultDockerNamespace = "library"
+	DefaultDockerTag       = "latest"
+)
+
+var (
+	RepoNameComponentRegexp = regexp.MustCompile(`^[a-z0-9]+(?:[._-][a-z0-9]+)*$`)
+)
+
+// NormalizeRef normalizes Docker image references of the following forms:
+//
+//   - [docker.io/][namespace/]repo[:tag]
+//   - [docker.io/][namespace/]repo[@digest]
+//
+// This is useful for the purposes of comparison and also in cases where a
+// canonical representation of a Docker Hub image reference is needed. Any reference
+// that cannot be normalized will return an error.
+//
+// Examples:
+//
+//	"nginx"                    -> "docker.io/library/nginx:latest"
+//	"user/repo:v1.0"           -> "docker.io/user/repo:v1.0"
+//	"docker.io/library/nginx"  -> "docker.io/library/nginx:latest"
+//	"nginx@sha256:..."         -> "docker.io/library/nginx@sha256:..."
+func NormalizeRef(ref string) (string, error) {
```

Comment:
This would be more consistent and more intuitively understood.

```suggestion
func NormalizeURL(repoURL string) (string, error) {
```